lifetimes/fitters/pareto_nbd_fitter.py

```diff
@@ -525,19 +528,64 @@ def _fit(
             if minimize_options["disp"]:
                 print("Optimize function with {}".format(fit_method))
 
-            output = minimize(
-                self._negative_log_likelihood,
-                method=fit_method,
-                tol=tol,
-                x0=current_init_params,
-                args=minimizing_function_args,
-                options=minimize_options,
-            )
+            # Convert printed numpy warnings to actual RuntimeWarnings
+            numpy.seterr('warn')
+            # Use python's warnings module to catch warnings as if they're errors,
+            # specifically warnings with an 'invalid value encountered in subtract'
+            # message, which is the warning that implies an infinite or nan
+            # log-likelihood value
+            warnings.filterwarnings('error', message='invalid value encountered in subtract')
+
+            stored_params = []
+            def print_and_store_params(xk):
+                print(xk)
+                stored_params.append(xk)
+
+            try:
+                output = minimize(
+                    self._negative_log_likelihood,
+                    method=fit_method,
+                    tol=tol,
+                    x0=current_init_params,
+                    callback=print_and_store_params,
+                    args=minimizing_function_args,
+                    options=minimize_options,
+                )
+            except RuntimeWarning:
+                print("Warning encountered, retrying...")
+                last_params = stored_params[-1]
+                f, r, t, weights, reg_param = minimizing_function_args
+                likelihoods = ParetoNBDFitter._conditional_log_likelihood(last_params, f, r, t)
+
+                inf_indices = np.where(np.isinf(likelihoods))
+                nan_indices = np.where(np.isnan(likelihoods))
+                bad_indices = np.concatenate((inf_indices[0], nan_indices[0]))
+                print(f"Number of bad indices: {bad_indices.size}")
+                print(f"Problematic entries: {f[bad_indices]}, {r[bad_indices]}, {t[bad_indices]}")
```

Comment:
These will be entirely numeric, right? Specifically, is there any chance we're logging PII here?

Reply:
correct, no PII